bgpd/bgp_fsm.c

```diff
@@ -1181,6 +1181,19 @@ int bgp_stop(struct peer *peer)
 		}
 	}
 
+	/* If a BGP speaker decides to de-configure a peer, then
```

Comment:
bgp_stop() would be called when "BGP_Stop" even is raised. And "BGP_Stop" event can be raised at different states of bgp and for various reasons, not only for "Peer De-configured".
Eg clear bgp cmd  

so we would end up 
1. sending "Peer De-configured" notification wrongly even when the peer is not delete. 
2. We might end up sending more than one notification message (pls see the below comment for the reason)